tsconfig.release.json

```diff
@@ -7,6 +7,7 @@
     "noUnusedLocals": true,
     "outDir": "lib",
     "stripInternal": true,
+    "skipLibCheck": true,
```

Comment:
q: what prompted this change?

Reply:
The Admin SDK updated the `@google-cloud/firestore` package which update the `google-gax` package and caused a nasty ts build error from the dependencies (https://github.com/firebase/firebase-functions/runs/6132285684?check_suite_focus=true), this change tries to silence that error.

Do you have a better suggestion on how to handle that?

Reply:
Yeah I'm a little worried that we are kicking the error down to the customer? Let's check to see if this error is legit worth fixing?

Reply:
So I've learned that:

We could get around this issue by updating `tsconfig.release.json`:

```json
{
  "lib": ["es2018", "dom"]
}
```

This means that we are including all type definitions for the `dom` lib while we compile our project. [gax-nodejs](https://github.com/googleapis/gax-nodejs/blob/main/tsconfig.json) and [nodejs-firestore](https://github.com/googleapis/nodejs-firestore/blob/main/tsconfig.json) seems to do this. (Surprisingly, Firebase Admin SDK doesn't do this yet still seems to be fine)

[`AbortController`](https://nodejs.org/api/globals.html#class-abortcontroller) was introduced 14.17.0. This means that by adding `dom` we are sorta kinda implictly saying that you must use Node v14 and onward in order to use our SDK? Still thinking whether this is fine for us to do or we should just go ahead and try to use the `skipLibCheck` escape hatch. 🤔

Reply:
Yeah I found that also but didn't know if we should be adding all those definitions. Maybe @inlined has some thoughts about this?

Reply:
Try removing `skipLibCheck` and re-add `node-fetch` to `package.json`. AFAICT it polyfills the missing types.

Reply:
@inlined build still fails with the same error

Reply:
thanks!